doc/source/whatsnew/v0.24.0.txt

```diff
@@ -177,7 +177,8 @@ Other Enhancements
 - :func:`read_html` copies cell data across ``colspan`` and ``rowspan``, and it treats all-``th`` table rows as headers if ``header`` kwarg is not given and there is no ``thead`` (:issue:`17054`)
 - :meth:`Series.nlargest`, :meth:`Series.nsmallest`, :meth:`DataFrame.nlargest`, and :meth:`DataFrame.nsmallest` now accept the value ``"all"`` for the ``keep`` argument. This keeps all ties for the nth largest/smallest value (:issue:`16818`)
 - :class:`IntervalIndex` has gained the :meth:`~IntervalIndex.set_closed` method to change the existing ``closed`` value (:issue:`21670`)
-- :func:`~DataFrame.to_csv` and :func:`~DataFrame.to_json` now support ``compression='infer'`` to infer compression based on filename (:issue:`15008`)
+- :func:`~DataFrame.to_csv`, :func:`~Series.to_csv`, :func:`~DataFrame.to_json`, and :func:`~Series.to_json` now support ``compression='infer'`` to infer compression based on filename (:issue:`15008`).
+  The default compression for ``to_csv``, ``to_json``, and ``to_pickle`` methods has been updated to ``'infer'`` (:issue:`22004`).
```

Comment:
can you add a tiny bit more detail that this means to infer based on the file extension.

any updates needed in io.rst?

Reply:
> can you add a tiny bit more detail that this means to infer based on the file extension.

Done in https://github.com/pandas-dev/pandas/pull/22011/commits/1ba8f3a281a71db33d8baa6a9a4c9fd7e40c625a

> any updates needed in io.rst?

I looked through https://pandas.pydata.org/pandas-docs/version/0.23.3/io.html and only saw the need for one small update (a `versionchanged` note) added in https://github.com/pandas-dev/pandas/pull/22011/commits/24e051e4d147d341de86e1919c74960a1b50253f.